pkg/server/server.go

```diff
@@ -236,11 +249,21 @@ func New(
 		}
 
 		if v := tuning.Snapshot; v != nil {
-			logrus.WithFields(logrus.Fields{"threshold": v.Threshold, "trailing": v.Trailing}).Print("Configure dqlite raft snapshot parameters")
-			options = append(options, app.WithSnapshotParams(dqlite.SnapshotParams{
-				Threshold: v.Threshold,
-				Trailing:  v.Trailing,
-			}))
+			logrus.WithFields(logrus.Fields{"threshold": v.Threshold, "trailing": v.Trailing}).Print("Initial dqlite raft snapshot parameters before adjustments")
+
+			if v.Trailing < minTrailing && v.Threshold > minThreshold {
+				snapshotParameters.Trailing = v.Threshold * scalingFactor
+				snapshotParameters.Threshold = v.Threshold
+			} else if v.Threshold < minThreshold && v.Trailing > minThreshold {
```

Comment:
I _think_ this should be:
```suggestion
			} else if v.Threshold < minThreshold && v.Trailing > minTrailing {
```